### ISSUE TITLE
Is oclgrind supporting generic address space?

### ISSUE THREAD

**Issue by ghostplant** [jrprice/Oclgrind#160]
I failed to execute the kernels if I remove `__global` tags for data pointers.

**Comment by jrprice** [comment]
No, Oclgrind does not currently support the generic address space. It currently has distinct address spaces for global, local and private memory, so some work would be needed to make it possible to identify the address space from the pointer value (probably at the same time as reworking the addressing system to support SVM).
